version: "3.3" 
  
services: 
  portainer: 
    image: portainer/portainer-ce 
    container_name: portainer 
    command: -H unix:///var/run/docker.sock 
    networks: 
      - red
    restart: always
    env_file:
      - .env
    security_opt:
      - no-new-privileges
    environment:
      - PUID=1001
      - PGID=1001
      - TZ=Europe/Madrid
      - LOG_LEVEL=info
    volumes: 
      - /var/run/docker.sock:/var/run/docker.sock:ro
      - /patch/to/data/docker/portainer:/data:rw 
    labels: 
      # portainer 
      - "traefik.enable=true" 
      - "traefik.http.routers.portainer.rule=Host(`domain`)" 
      - "traefik.http.routers.portainer.entrypoints=websecure" 
      - "traefik.http.routers.portainer.tls=true" 
      - "traefik.http.routers.portainer.tls.certresolver=lets-encrypt" 
      - "traefik.http.services.portainer.loadbalancer.server.port=9443" 
      - "traefik.http.middlewares.portainer.headers.sslredirect=true"
      - "traefik.http.routers.portainer.middlewares=secured"
      - "traefik.http.middlewares.secured.chain.middlewares=portainer,lista-blanca"
      - "traefik.http.middlewares.lista-blanca.ipwhitelist.sourcerange=192.168.1.0/24, 10.13.13.1/24"

volumes: 
  portainer: 
    driver: local 
     
networks: 
  red: 
    external: true 
